x/wasm/internal/keeper/genesis.go

```diff
@@ -0,0 +1,74 @@
+package keeper
+
+import (
+	"bytes"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmwasm/wasmd/x/wasm/internal/types"
+	// authexported "github.com/cosmos/cosmos-sdk/x/auth/exported"
+	// "github.com/cosmwasm/wasmd/x/wasm/internal/types"
+)
+
+// InitGenesis sets supply information for genesis.
+//
+// CONTRACT: all types of accounts must have been already initialized/created
+func InitGenesis(ctx sdk.Context, keeper Keeper, data types.GenesisState) {
+	for id, info := range data.CodeInfos {
+		bytecode := data.CodesBytes[id]
+		newId, err := keeper.Create(ctx, info.Creator, bytecode)
+		if err != nil {
+			panic(err)
+		}
+		newInfo := keeper.GetCodeInfo(ctx, newId)
+		if !info.Creator.Equals(newInfo.Creator) {
+			panic("code creators not same")
+		}
+		if bytes.Compare(info.CodeHash, newInfo.CodeHash) != 0 {
```

Comment:
This definitely is super important to validate :+1:

I would just recommend `! bytes.Equal(.., ..)` instead for readability